src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/controller/PartnershipController.java

```diff
@@ -62,4 +70,39 @@ public ResponseEntity<Object> createPartnership(
             return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+
+    @PatchMapping("/{" + URL_PARAM_SUBMISSION_ID + "}")
+    public ResponseEntity<Object> updatePartnership(
+            @RequestAttribute(TRANSACTION_KEY) Transaction transaction,
+            @PathVariable(URL_PARAM_SUBMISSION_ID) String submissionId,
+            @RequestBody Map<String, Object> body,
+            @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId,
+            @RequestHeader(value = ERIC_IDENTITY) String userId
+    ) throws JsonProcessingException {
+
+        String transactionId = transaction.getId();
+        HashMap<String, Object> logMap = new HashMap<>();
+        logMap.put(URL_PARAM_TRANSACTION_ID, transactionId);
+
+        try {
+            String type = (String) body.get("type");
+            var dataType = DataType.valueOf(type.toUpperCase());
+            final Map<String, Object> data = extractData(body);
```

Comment:
could you catch the JsonProcessingException in this try and handle it the same as the ServiceException?

Reply:
need more explanation about what you want here

Reply:
updated